libraries/AP_Filesystem/AP_Filesystem_Sys.h

```diff
@@ -17,6 +17,8 @@
 
 #include "AP_Filesystem_backend.h"
 
+class ExpandingString;
```

Comment:
Seems a little suspect to forward declare something that's actually used by the class

Reply:
it is so we can avoid having this header everywhere